lib/robots/dor_repo/gis_assembly/assign_placenames.rb

```diff
@@ -11,73 +11,78 @@ def initialize
         def perform_work
           logger.debug "assign-placenames working on #{bare_druid}"
 
-          rootdir = GisRobotSuite.locate_druid_path bare_druid, type: :stage
-          mods_filename = File.join(rootdir, 'metadata', 'descMetadata.xml')
-          raise "assign-placenames: #{bare_druid} is missing MODS metadata" unless File.size?(mods_filename)
+          resolve_placenames
 
-          resolve_placenames(mods_filename)
-          raise "assign-placenames: #{bare_druid} corrupted MODS metadata" unless File.size?(mods_filename)
+          description_props = Cocina::Models::Mapping::FromMods::Description.props(mods: mods_doc, druid:,
+                                                                                   label: cocina_object.label)
+          object_client.update(params: cocina_object.new(description: description_props))
         end
 
         private
 
+        def mods_doc
```

Comment:
It might be useful to leave a TODO note in the places where we are still converting to MODS in memory so we can clean them up later?